compiler/javac/src/org/jetbrains/kotlin/wrappers/symbols/JavacType.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.wrappers.symbols
+
+import org.jetbrains.kotlin.javac.Javac
+import org.jetbrains.kotlin.load.java.structure.JavaAnnotation
+import org.jetbrains.kotlin.load.java.structure.JavaAnnotationOwner
+import org.jetbrains.kotlin.load.java.structure.JavaType
+import org.jetbrains.kotlin.name.FqName
+import javax.lang.model.type.TypeKind
+import javax.lang.model.type.TypeMirror
+
+open class JavacType<out T : TypeMirror>(val typeMirror: T,
+                                         val javac: Javac) : JavaType, JavaAnnotationOwner {
+
+    companion object {
+        fun <TM : TypeMirror> create(t: TM, javac: Javac) = when {
+            t.kind.isPrimitive || t.toString() == "void" -> JavacPrimitiveType(t, javac)
```

Comment:
`PrimitiveType` includes `boolean`, `byte`, `short`, `int`, `long`, `char`, `float`, and `double`.